extpsdk/README.md

```diff
@@ -0,0 +1,260 @@
+ 
+This context parameter is to be used in the various query response calls.
+
+For the handle_connect, handle_query, and handle_publish calls, there will b be a second dict parameter. This will contain the message sent from Vantiq to the connector (source).
+
+Note that the handlers all return an _Awaitable_.  This means that they can operate using _asyncio_ to do other things.  However, _asyncio_ operations should be careful not to block the event loop. If an handler is going to take a long time to run, you will want to set it running separately.
+
+#### <a name="connect" id="connect"></a>Connection (Configuration)
+
+The `handle_connect` callback is called as part of the Vantiq connection sequence, delivering the source configuration to the connector. receives the message containing a configuration document created on the Vantiq client.
+
+This callback will be called with two parameter: the context and the configuration document.
+The _context_ parameter was outlined in the previous section.
+The _configuration_ document is a `dict` object containing the source's configuration document. There is no format specification of the configuration document -- its content is defined by the connector.  Generally, the configuration document allows the connector to be tailored to specific purposes.  It may contain database connection information, machine learning models to run, or operational information like cache sizes, etc.
+
+##### Reconnection
+Under certain circumstances (_e.g._, a change to the source's configuration document or some network connection issue), the Vantiq server may disconnect
+you from the source with the intent that you will immediately reconnect.
+
+This operation is unavoidable, but is generally transparent to the connector.  When this happens, the [`handle_close` callback](#closeHandler) will be called, and then the
+[`handle_connect` callback](#closeHandler).  Since this may be an indication that the source configuration has changed, each _connect_ call should be treated as a new connection/configuration.
+
+This process is automatic.  No special work by the connector is required.
+
+### Processing Messages
+
+Messages involving a source come in three (3) flavors:
+
+* Messages (events) sent by a source
+* Query messages sent to a source. Query messages, in turn, have responses.
+* Messages published to a source
+
+The following sections outline the work involved here.
+
+#### Notifications
+
+Notifications are messages that the source will pass on to Vantiq rules that include 
+`WHEN EVENT OCCURS ON "/sources/<source name>"`. To send one, simply call
+`Vantiq.send_notification(<message to be sent>)`, which will translate the message into JSON and add everything Vantiq needs to recognize the message.  The _message to be sent_ shoud be a `dict` object.
```

Comment:
Should this be `VantiqSourceConnection.send_notification(<message to be sent>)`?